src/amqp.js

```diff
@@ -0,0 +1,340 @@
+    }
+  }
+
+  // MARK: Ack messages
+  /**
+   * Ack messages
+   *
+   * @param {Array[Object]} messages Array of message objects
+   * @returns {void} returns nothing
+   */
+  async function ackMessages(messages) {
+    messages.forEach(message => {
+      debugDev(`Ack message ${message.properties.correlationId}`);
+      channel.ack(message);
+    });
+    await setTimeoutPromise(10);
+    return;
+  }
+
+  // MARK: Nack messages
+  /**
+   * Nack messages
+   *
+   * @param {Array[Object]} messages Array of message objects
+   * @returns {void} returns nothing
+   */
+  async function nackMessages(messages) {
+    messages.forEach(message => {
+      debugDev(`Nack message ${message.properties.correlationId}`);
+      channel.nack(message, false, true);
+    });
+    await setTimeoutPromise(10);
+    return;
+  }
+
+  // MARK: Send to queue
+  /**
+   * Sends to queue
+   * Send to queue uses blob id and status to form queue id.
+   * E.g. 905e283f-2644-4d26-8432-6634fbbc0161.PENDING_LOOKUPS
+   *
+   * @param {Object} params object: blobId, status, data
+   * @param {uuid} params.blobId BlobId for blob to be handled
+   * @param {string} params.status Status for queue
+   * @param {object} params.data Data object sent to queue
+   * @returns {boolean} returns true on success
+   */
+  async function sendToQueue({blobId, status, data}) {
+    const queue = generateQueueId({blobId, status});
+    debug(`sendToQueue`);
+    // eslint-disable-next-line no-useless-catch
+    try {
+      debug(`Queue ${queue}`);
+      debugData(`Data ${JSON.stringify(data)}`);
+
+      errorUndefinedQueue(queue);
+
+      debugDev(`Asserting queue: ${queue}`);
+      await channel.assertQueue(queue, {durable: true});
+
+      debugDev(`Actually sendToQueue: ${queue}`);
+      await channel.sendToQueue(
+        queue,
+        Buffer.from(JSON.stringify({data})),
+        {
+          correlationId: blobId,
+          persistent: true
+        }
+      );
+      debugDev(`Send message to queue: ${queue}`);
+
+      return true;
+    } catch (error) {
+      const errorToThrow = error;
+      //if (error instanceof ApiError) {
+      //  throw error;
+      //}
+      debug(`SendToQueue errored: ${JSON.stringify(error)}`);
+      handleAmqpErrors(errorToThrow);
+    }
+  }
+
+  async function removeQueue({blobId, status}) {
+    const queue = generateQueueId({blobId, status});
+    // deleteQueue borks the channel if the queue does not exist
+    // -> use throwaway tempChannel to avoid killing actual channel in use
+    // this might be doable also with assertQueue before deleteQueue
+    const tempChannel = await connection.createChannel();
+    debug(`Removing queue ${queue}.`);
+    await tempChannel.deleteQueue(queue);
+
+    if (tempChannel) {
+      await tempChannel.close();
+      return;
+    }
+
+    return;
+  }
+
+  // ----------------
+  // Helper functions
+  // ----------------
+
+  function checkMessageCount(channelInfo) {
```

Comment:
Function name should tell what is inspected - proposing renaming to "channelContainsMessages" or similar.